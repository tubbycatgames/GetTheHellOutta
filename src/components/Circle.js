import styled from "styled-components/native";

const getDiameter = props => props.diameter;

const Circle = styled.View`
  background-color: white;
  border-color: #ccc;
  border-radius: ${getDiameter};
  border-width: 4;
  height: ${getDiameter};
  left: ${props => props.left};
  position: absolute;
  top: ${props => props.top};
  width: ${getDiameter};
`;

export default Circle;
