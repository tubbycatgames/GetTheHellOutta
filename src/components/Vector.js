export default class Vector {
  static zero = new Vector(0, 0);
  static one = new Vector(1, 1);

  static add(v1, v2) {
    return new Vector(v1.x + v2.x, v1.y + v2.y);
  }

  static mult(v, factor) {
    return new Vector(v.x * factor, v.y * factor);
  }

  constructor(_x, _y) {
    this.x = _x;
    this.y = _y;
  }
}
