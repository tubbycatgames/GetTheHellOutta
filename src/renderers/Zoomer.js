import PropTypes from "prop-types";
import React from "react";

import { MediumCircle, Vector } from "../components";

const Zoomer = ({ createdAt, position }) => (
  <MediumCircle color={"#123456"} createdAt={createdAt} position={position} />
);

const vectorProp = PropTypes.instanceOf(Vector);

Zoomer.propTypes = {
  createdAt: PropTypes.number.isRequired,
  position: vectorProp.isRequired,
  velocity: vectorProp.isRequired
};

export default Zoomer;
