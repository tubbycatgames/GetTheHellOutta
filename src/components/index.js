import { MediumCircle } from "./Circle";
import Vector from "./Vector";

export { MediumCircle, Vector };
