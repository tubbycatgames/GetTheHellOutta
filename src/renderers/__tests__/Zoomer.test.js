import React from "react";
import renderer from "react-test-renderer";

import Zoomer from "../Zoomer";
import { Vector } from "../../components";

describe("Zoomer", () => {
  it("should render a zoomer", () => {
    const zoomer = renderer.create(
      <Zoomer
        color={"#123456"}
        createdAt={5}
        position={new Vector(21, 21)}
        velocity={Vector.zero}
      />
    );
    expect(zoomer.toJSON()).toMatchSnapshot();
  });
});
