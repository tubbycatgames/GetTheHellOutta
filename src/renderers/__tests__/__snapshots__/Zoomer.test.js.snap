// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Zoomer should render a zoomer 1`] = `
<View
  color="#123456"
  createdAt={5}
  position={
    Vector {
      "x": 21,
      "y": 21,
    }
  }
  radius={20}
  style={
    Array [
      Object {
        "backgroundColor": "#123456",
        "borderColor": "white",
        "borderRadius": 40,
        "borderWidth": 4,
        "height": 40,
        "left": 1,
        "position": "absolute",
        "top": 1,
        "width": 40,
      },
    ]
  }
/>
`;
